File: test/csharp/SpecGateFixtures/Conformance/Stateful/SetupWithParams/setup_with_params.cs
using SpecGate.Annotations;
namespace SpecGateFixtures.Conformance.Stateful.SetupWithParams;

/// <summary>
/// State machine whose setup takes a parameter, verifying that setup inputs
/// configure the initial state (here, the counter's starting value).
/// </summary>
public class Counter
{
    /// <summary>The current count; each assignment is captured as a state mutation.</summary>
    [SpecEvent("count")]
    public int Count { get; set; }

    /// <summary>Builds a counter starting at <paramref name="initial"/>.</summary>
    /// <param name="initial">The starting count (spec input <c>initial</c>).</param>
    /// <returns>A new <see cref="Counter"/> with <see cref="Count"/> = <paramref name="initial"/>.</returns>
    [SpecSetup("increment", Spec = "fixture.setup_with_params")]
    public static Counter Make([SpecInput("initial")] int initial) => new() { Count = initial };

    /// <summary>Increments the counter by one.</summary>
    [SpecOperation("increment", Spec = "fixture.setup_with_params")]
    public void Increment() => Count += 1;
}

File: test/rust/crates/specgate-fixtures/src/conformance/stateful/setup_with_params.rs
// Setup with input parameter — initial count is configurable.
use specgate::*;

#[spec_setup("increment", spec = "fixture.setup_with_params")]
pub fn make_counter(initial: i32) -> Counter {
    Counter { count: initial }
}

#[derive(SpecEvent)]
#[spec_component("fixture.setup_with_params")]
pub struct Counter {
    #[spec_event]
    pub count: i32,
}

impl Counter {
    #[spec_operation("increment", spec = "fixture.setup_with_params")]
    pub fn increment(&mut self) {
        self.count += 1;
    }
}
